File: core/src/main/java/com/hgianastasio/biblioulbrav2/core/search/exceptions/EmptyResultsException.java
package com.hgianastasio.biblioulbrav2.core.search.exceptions;

import com.hgianastasio.biblioulbrav2.core.search.models.SearchResult;

/**
 * Created by heitorgianastasio on 4/25/17.
 */

public class EmptyResultsException extends RuntimeException {
    private SearchResult result;

    public EmptyResultsException(SearchResult result) {
        this.result = result;
    }

    public EmptyResultsException(String s, SearchResult result) {
        super(s);
        this.result = result;
    }

    @Override
    public String getMessage() {
        return "Pesquisa sem resultados";
    }

    public SearchResult getResult() {
        return result;
    }
}

File: core/src/main/java/com/hgianastasio/biblioulbrav2/core/search/exceptions/EmptyResultsException.kt
package com.hgianastasio.biblioulbrav2.core.search.exceptions

import com.hgianastasio.biblioulbrav2.core.search.models.SearchResult

/**
 * Created by heitorgianastasio on 4/25/17.
 */
class EmptyResultsException : RuntimeException {
    var result: SearchResult
        private set

    constructor(result: SearchResult) {
        this.result = result
    }

    constructor(s: String?, result: SearchResult) : super(s) {
        this.result = result
    }

    override val message: String
        get() = "Pesquisa sem resultados"

}